.github/workflows/bench_pr.yml

```diff
@@ -80,14 +84,27 @@ jobs:
           aws-region: us-west-2
           role-to-assume: arn:aws:iam::083632199359:role/gha_oidc_risc0_cache_public_access
 
-      - name: Download previous benchmark data
-        run: aws s3 cp $S3_BENCH_PATH/external.json ./cache/external.json
-
       - run: cargo run --bin cargo-risczero --no-default-features -- risczero install --version $RISC0_TOOLCHAIN_VERSION
 
       - name: Run benchmarks
         run: cargo bench --bench fib -F $FEATURE
 
+      - name: Check for Existing Benchmark Data
+        run: |
+          aws s3 ls $S3_BENCH_PATH/external.json || NOT_EXIST=true
+          if [ $NOT_EXIST ]; then
+              echo "No ${{ matrix.os }}-${{ matrix.device }} Benchmark data yet... Will be uploaded..." && echo "NEW_BENCHMARK=true" >> $GITHUB_ENV
+          else
+              echo "Previous Benchmark data for ${{ matrix.os }}-${{ matrix.device }} exists... Continuing..."
+          fi
+
+      - name: (New Benchmarks) Upload Initial Benchmark Data
+        if: env.NEW_BENCHMARK == 'true'
+        run: aws s3 cp target/hotbench/fib/benchmark.json $S3_BENCH_PATH/external.json
+
+      - name: Download previous benchmark data
+        run: aws s3 cp $S3_BENCH_PATH/external.json ./cache/external.json
+
```

Comment:
@flaub - Added so all `bench_pr.yml` steps will work with new Benchmarks by uploading a baseline file to S3 if none exists. If a file does not exist in AWS, then the `Analyze benchmark results` job will fail since nothing was moved from an S3 bucket to `./cache/external.json` in a previous step, which is needed for the comparison.

New steps:
1. Verify a `$S3_BENCH_PATH/external.json` file exists for the `{{ matrix.os }}-${{ matrix.device }}` benchmark being tested.
2. If JSON file exists, it skips the second step to upload initial data.
3. If no JSON exists, it uploads the file generated in the `Run benchmarks` step as a baseline going forward.

An alternative process to putting this in CI would be running the benchmark locally, creating a new folder in S3 with the proper path, then manually moving the local benchmark file there with the name `external.json`. All things considered, keeping it in CI would keep things simpler and more consistent.

---

The jobs ran earlier today as expected with the `g6.xlarge` instances:

- Run 1:
  - Ran the `(New Benchmarks) Upload Initial Benchmark Data` step and successfully uploaded a baseline JSON file in the correct path 
  - https://github.com/risc0/risc0/actions/runs/9521185195/job/26248070239

- Run 2:
  - Skipped the `(New Benchmarks) Upload Initial Benchmark Data` step since a baseline file was found after being generated from the previous run.
  - https://github.com/risc0/risc0/actions/runs/9521185195/job/26251450354?pr=1974

Reply:
This will be removed from this PR since the `g6.xlarge` runners will only be added to `bench_reports.yml` and 'bench_trends.yml`. 

It will be added to documentation for temporary use in tests of new all instances and to generate the initial `external.json` file for devices that will also be added to `bench_pr.yml`.